app/assets/javascripts/scp-dot-plot.js

```diff
@@ -0,0 +1,296 @@
+
+    // Calling % 27 will always return to the beginning of colorBrewerSet once we use all 27 values
+    $(sortedAnnots).each(function(index, annot) {
+      annotColorModel[selectedAnnot][annot] = colorBrewerSet[index % 27];
+    });
+    config.columnColorModel = annotColorModel;
+  }
+
+  config.colorScheme = dotPlotColorScheme;
+
+  // Log dot plot initialization in Google Analytics
+  if (typeof window.dotPlot === 'undefined') {
+    // Consistent with e.g. IGV, Ideogram
+    ga('send', 'event', 'dot-plot', 'initialize');
+  }
+
+  // Instantiate dot plot and embed in DOM element
+  window.dotPlot = new morpheus.HeatMap(config);
+  window.dotPlot.tabManager.setOptions({autohideTabBar: true});
+  $(target).off();
+  $(target).on('heatMapLoaded', function (e, heatMap) {
+
+    // Remove verbose tab atop Morpheus dot plot
+    var tabItems = dotPlot.tabManager.getTabItems();
+    window.dotPlot.tabManager.setActiveTab(tabItems[1].id);
+    window.dotPlot.tabManager.remove(tabItems[0].id);
+
+    renderDotPlotLegend();
+
+    // Remove "Options" toolbar button until legend can be updated upon
+    // changing default options for size and color (SCP-1738).
+    // setTimeout is a kludge, but seemingly the only way to do this.
+    setTimeout(function() {
+      var options = $('[data-action="Options"]');
+      options.next('.morpheus-button-divider').remove();
+      options.remove();
+      console.log('removed options')
+    }, 50);
+  });
+
+  // Set render variable to true for tests
+  $(target).data('morpheus', dotPlot);
+  $(target).data('rendered', true);
+  console.log('render status of ' + target + ' at end: ' + $(target).data('rendered'));
+}
+
+function drawDotplot(height) {
+  $(window).off('resizeEnd');
+
+  // Clear out previous stored dotplot object
+  $('#dot-plot').data('dotplot', null);
+
+  // If height isn't specified, pull from stored value, defaults to 500
+  if (height === undefined) {
+    height = $('#dot-plot').data('height');
+  }
+
+  // Pull fit type as well, defaults to ''
+  var fit = $('#dot-plot').data('fit');
+
+  var selectedAnnot = $('#annotation').val();
+  var annotName = selectedAnnot.split('--')[0];
+  var annotType = selectedAnnot.split('--')[1];
+
+  var cluster = $('#cluster').val();
+  $('#search_cluster').val(cluster);
+  $('#search_annotation').val(''); // clear value first
+  $('#search_annotation').val(selectedAnnot);
+
+  var newAnnotPath = dotPlotAnnotPathBase + '?cluster=' + cluster + '&annotation=' + selectedAnnot + '&request_user_token=' + requestToken;
+
+  var renderUrlParams = getRenderUrlParams();
+  // Get annotation values to set color values in Morpheus and draw dotplot in callback
+  $.ajax({
+    url: dotPlotAnnotValuesPath + '?' + renderUrlParams,
+    dataType: 'JSON',
+    success: function(annotations) {
+      renderMorpheusDotPlot(dataPath, newAnnotPath, annotName, annotType, '#dot-plot', annotations, fit, height);
+    }
+  });
+}
+
+$('#annotation').change(function() {
```

Comment:
How is this not clobbering the event listener that's already in the page? See https://github.com/broadinstitute/single_cell_portal_core/blob/master/app/views/site/_study_visualize.html.erb#L102 and https://github.com/broadinstitute/single_cell_portal_core/blob/master/app/views/site/_view_gene_expression_heatmap.html.erb#L103 for reference.  I tried it out locally and it's doing the right thing, but looking at the code it seems to me that it shouldn't (unless I'm missing something intrinsic about event listeners).

Reply:
Good catch!  It is clobbering the event listener.

However, it turns out none of the event listeners  in `scp-dot-plot.js` are actually used, and their functionality is all handled in the event listeners attached via `_view_gene_expression_heatmap.html.erb`.

So I've removed that dead code.  It looks like it was started as a way to do more refined event handling before I moved dot-plot-specific code out of `_view_gene_expression_heatmap.html.erb`, but the refinement was never completed.  In other words, the event handler in https://github.com/broadinstitute/single_cell_portal_core/blob/master/app/views/site/_view_gene_expression_heatmap.html.erb#L103 renders both dot plots and heatmaps when an annotation changes.  The unrefined event handler there is chattier than strictly necessary, but seems fine.